build-tools-internal/src/integTest/java/org/elasticsearch/gradle/jarhell/ThirdPartyAuditTaskIT.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.jarhell;
+
+import org.elasticsearch.gradle.internal.test.GradleIntegrationTestCase;
+import org.gradle.testkit.runner.BuildResult;
+import org.junit.Before;
+
+import static org.elasticsearch.gradle.internal.test.TestClasspathUtils.setupJarJdkClasspath;
+
+public class ThirdPartyAuditTaskIT extends GradleIntegrationTestCase {
+
+    @Override
+    public String projectName() {
+        return "thirdPartyAudit";
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        // Build the sample jars
+        getGradleRunner().withArguments(":sample_jars:build", "-s").build();
+        // propagate jdkjarhell jar
+    }
+
+    public void testElasticsearchIgnored() {
+        BuildResult result = getGradleRunner().withArguments(
+            ":clean",
+            ":empty",
+            "-s",
+            "-PcompileOnlyGroup=elasticsearch.gradle:broken-log4j",
+            "-PcompileOnlyVersion=0.0.1",
+            "-PcompileGroup=elasticsearch.gradle:dummy-io",
+            "-PcompileVersion=0.0.1"
+        ).build();
+        assertTaskNoSource(result, ":empty");
+        assertNoDeprecationWarning(result);
+    }
+
+    public void testViolationFoundAndCompileOnlyIgnored() {
+        setupJarJdkClasspath(getProjectDir());
+
+        BuildResult result = getGradleRunner().withArguments(
+            ":clean",
+            ":absurd",
+            "-s",
+            "-PcompileOnlyGroup=other.gradle:broken-log4j",
+            "-PcompileOnlyVersion=0.0.1",
+            "-PcompileGroup=other.gradle:dummy-io",
+            "-PcompileVersion=0.0.1"
+        ).buildAndFail();
+
+        assertTaskFailed(result, ":absurd");
+        assertOutputContains(result.getOutput(), "Classes with violations:", "  * TestingIO", "> Audit of third party dependencies failed");
+        assertOutputMissing(result.getOutput(), "Missing classes:");
+        assertNoDeprecationWarning(result);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 46:61, 65:84.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.